### PR TITLE
docs: how to run a node

### PR DESCRIPTION
**Pull request by samricotta**
## Summary

For bbn-test-5 one of the guides we will have is node. This is closely aligned with validator. 

There are still a few changes but mostly finshed

### CODE REVIEW

File: bbn-test-5/babylon-node/README.md [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+```shell
+git clone git@github.com:babylonlabs-io/babylon.git
+cd babylon
+git checkout bbn-test-5
+make install
+```
+<!-- TODO: testnet tag to be defined -->
+This command does the following:
+- Builds the daemon
+- Compiles all the Go packages in the project
```
Comment (filippos47) [inline bbn-test-5/babylon-node/README.md:33]:
```suggestion
```
I'd remove this line, it's covered by the build.

File: bbn-test-5/babylon-node/README.md [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+This command will generate several important configuration files 
+including `app.toml`, `client.toml`, and `genesis.json`:
+
+```shell
+babylond init <moniker> --chain-id bbn-test-5 --home <path>
+```
+
+Parameters:
+- `<moniker>`: a unique identifier for your node for example `node0`
+- `--home`: *optional* flag that specifies the directory where your
```
Comment (filippos47) [inline bbn-test-5/babylon-node/README.md:65]: Should we mention the default homedir?

File: bbn-test-5/babylon-node/README.md [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+network = "signet" # The Babylon testnet connects to the signet Bitcoin network
+```
+
+Parameters:
+- `minimum-gas-prices`: The minimum gas price (in this example we use `0.005ubbn`)
+   that your node will accept for transactions. Transactions with lower gas 
+   prices will be rejected.
+- `iavl-cache-size`: Set to 0 to disable the IAVL tree caching. This reduces 
+   memory usage.
+- `iavl-disable-fastnode`: Set to true to disable the fast node feature.
```
Comment (filippos47) [inline bbn-test-5/babylon-node/README.md:94]:
```suggestion
- `iavl-disable-fastnode`: Set to true to disable the fast node feature. This
   reduces memory usage.
```

File: bbn-test-5/babylon-node/README.md [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+the initial state of the blockchain and is crucial for successfully syncing 
+your node. You can download it from either:
+
+- The Babylon Networks repository: [bbn-test-5](../genesis.tar.bz2)
+- Directly using these commands:
+```shell
+wget https://github.com/babylonlabs-io/networks/raw/main/bbn-test-5/genesis.tar.bz2 
+tar -xjf genesis.tar.bz2 && rm genesis.tar.bz2
+mv genesis.json <path>/config/genesis.json # You must insert the home directory of your node
+```
```
Comment (filippos47) [inline bbn-test-5/babylon-node/README.md:124]: A bit duplicated. I'd say "you can inspect the file [here](link) and use the following commands to download it directly"

File: bbn-test-5/babylon-node/README.md [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+
+Additionally, verify that the `chain-id` in the genesis file matches the one used in 
+your initialization command (`bbn-test-5`). This ensures your node connects 
+to the correct network.
```
Comment (filippos47) [inline bbn-test-5/babylon-node/README.md:128]: We give them instructions on how to get the genesis file, I don't think that's needed.
Also, this evokes the question "how?".

File: bbn-test-5/babylon-node/README.md [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+
+## 3. Prepare for sync
+<!-- TODO: Specify height and version -->
+Before starting your node sync, it's important to understand that Testnet-5
```
Comment (filippos47) [inline bbn-test-5/babylon-node/README.md:132]:
```suggestion
Before starting your node sync, it's important to note that Testnet-5 
```

File: bbn-test-5/babylon-node/README.md [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+There are three options you can choose from when syncing:
+1. Sync through a network snapshot (fastest method)
+2. Sync through state sync (quick catch-up without full history)
+3. Sync from scratch (complete sync from block 1)
+
+If you're syncing from scratch, you'll need to first sync with v0.9.0 
+until height `X`, then upgrade your binary to `vA.B.C` to continue syncing. 
+Alternatively, you can use state sync or a snapshot from after the upgrade 
+height to start directly with `vA.B.C`. Below are several strategies to sync 
+your node, choose the one that best fits your needs.
```
Comment (filippos47) [inline bbn-test-5/babylon-node/README.md:146]: I don't think we need this.
You already explain the methods above and expand on them below.

**Review by filippos47 — DISMISSED** [review]
Very nice work!

**Review by vitsalis — COMMENTED** [review]
My comments. The guide is being overly specific with block heights, tags, etc. significantly harming it's maintenability. A single change or a single software upgrade will break the entire guide. I recommend we keep it general in this guide, and point the user to the networks homepage to get the latest information on upgrades, binaries, etc. This way, we'll only have to maintain a single page with this info and won't have to update this guide each time.

File: bbn-test-5/babylon-node/README.md [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+go version
+```
+
+Next, clone the Babylon codebase
+and install the Babylon binary:
+
+```shell
+git clone git@github.com:babylonlabs-io/babylon.git
+cd babylon
+git checkout <tag>
```
Comment (vitsalis) [inline bbn-test-5/babylon-node/README.md:32]:
```suggestion
# tag corresponds to the version of the software
# you want to install -- depends on which
# height you sync from
git checkout <tag>
```

File: bbn-test-5/babylon-node/README.md [hunk 9]
```diff
@@ -0,0 +1,237 @@
...
+- `--chain-id`: The chain ID of the Babylon chain you connect to
+- `--home`: *optional* flag that specifies the directory where your 
+   node files will be stored, for example `--home ./nodeDir`
+   The default home directory for your Babylon node is:
+   - Linux/Mac: `~/.babylond/`
+   - Windows: `%USERPROFILE%\.babylond\`
+
+After initialization, you'll need to modify the following configuration files:
+
+1. On `app.toml`, update these the following settings:
```
Comment (vitsalis) [inline bbn-test-5/babylon-node/README.md:78]:
```suggestion
1. On `app.toml`, update the following settings:
```

File: bbn-test-5/babylon-node/README.md [hunk 10]
```diff
@@ -0,0 +1,237 @@
...
+following commands to download it directly:
+
+```shell
+wget https://github.com/babylonlabs-io/networks/raw/main/bbn-test-5/genesis.tar.bz2 
+tar -xjf genesis.tar.bz2 && rm genesis.tar.bz2
+mv genesis.json <path>/config/genesis.json # You must insert the home directory of your node
+```
+
+## 3. Prepare for sync
+Before starting your node sync, it's important to that the initial release at
```
Comment (vitsalis) [inline bbn-test-5/babylon-node/README.md:135]:
```suggestion
Before starting your node sync, it's important to note that the initial release at 
```

File: bbn-test-5/babylon-node/README.md [hunk 11]
```diff
@@ -0,0 +1,237 @@
...
+
+Once configured, proceed to [Start the node](#4-start-the-node).
+
+### 3.3. Sync from scratch
+
+Lastly, you can also sync from scratch, i.e., sync from block `1`. Syncing from 
+scratch means downloading and verifying every block from the beginning 
+of the blockchain (genesis block) to the current block.
+
+This will require you to use 2 different babylond binaries and perform the
```
Comment (vitsalis) [inline bbn-test-5/babylon-node/README.md:202]: not just two, one for each version upgradee

File: bbn-test-5/babylon-node/README.md [hunk 12]
```diff
@@ -0,0 +1,237 @@
...
+
+Lastly, you can also sync from scratch, i.e., sync from block `1`. Syncing from 
+scratch means downloading and verifying every block from the beginning 
+of the blockchain (genesis block) to the current block.
+
+This will require you to use 2 different babylond binaries and perform the 
+babylon software upgrade when needed.
+
+1. First, follow the installation steps in [Section 1](#1-install-babylon-binary), 
+but use tag `v0.9.0` instead of `bbn-test-5`.
```
Comment (vitsalis) [inline bbn-test-5/babylon-node/README.md:206]:
```suggestion
1. First, follow the installation steps in [Section 1](#1-install-babylon-binary)
using the genesis software version `v0.9.0` in place of `<tag>`. 
```

File: bbn-test-5/babylon-node/README.md [hunk 13]
```diff
@@ -0,0 +1,237 @@
...
+Lastly, you can also sync from scratch, i.e., sync from block `1`. Syncing from 
+scratch means downloading and verifying every block from the beginning 
+of the blockchain (genesis block) to the current block.
+
+This will require you to use 2 different babylond binaries and perform the 
+babylon software upgrade when needed.
+
+1. First, follow the installation steps in [Section 1](#1-install-babylon-binary), 
+but use tag `v0.9.0` instead of `bbn-test-5`. 
+<!-- TODO: Update tag -->
```
Comment (vitsalis) [inline bbn-test-5/babylon-node/README.md:207]: No need for this TODO

File: bbn-test-5/babylon-node/README.md [hunk 14]
```diff
@@ -0,0 +1,237 @@
...
+2. Start your node as specified in section [Start the node](#4-start-the-node).
+
+Your node will sync blocks until it reaches the specified upgrade height.
+
+<!-- TODO: Add log -->
+ <!-- TODO: Specify version -->
+3.At this point, return to [Section 1](#1-install-babylon-binary) and install 
+version `vA.B.C`.
+
+4. Restart your node. Your node will then start syncing the rest of the blocks
```
Comment (vitsalis) [inline bbn-test-5/babylon-node/README.md:218]:
```suggestion
2. Start your node as specified in section [Start the node](#4-start-the-node).

Your node will sync blocks until it reaches an upgrade height.
At that point, you will have to get the new software version
defined by that height, and go back to step (1) in order
to install it and restart.

You will have to repeat the above two steps until you sync
with the full blockchain.
```

**Review by vitsalis — APPROVED** [review]

File: bbn-test-5/babylon-node/README.md [hunk 15]
```diff
@@ -0,0 +1,237 @@
...
+3. [Prepare for sync](#3-prepare-for-sync)
+   1. [Sync through a network snapshot](#31-sync-through-a-network-snapshot)
+   2. [Sync through state sync](#32-sync-through-state-sync)
+   3. [Sync from scratch](#33-sync-from-scratch)
+4. [Start the node](#4-start-the-node)
+
+## 1. Install Babylon Binary 
+
+Installing the Babylon binary requires
+a Golang installation.
```
Comment (Lazar955) [inline bbn-test-5/babylon-node/README.md:16]: nit: why a line break here

File: bbn-test-5/babylon-node/README.md [hunk 16]
```diff
@@ -0,0 +1,237 @@
...
+
+1. On `app.toml`, update these the following settings:
+
+```shell
+# Base configuration
+# Minimum gas prices that this node will accept
+minimum-gas-prices = "0.005ubbn"
+
+iavl-cache-size = 0
+iavl-disable-fastnode=true
```
Comment (Lazar955) [inline bbn-test-5/babylon-node/README.md:86]:
```suggestion
iavl-cache-size = 5000 # will result in 3GB of memory usage
iavl-disable-fastnode=false
```

File: bbn-test-5/babylon-node/README.md [hunk 17]
```diff
@@ -0,0 +1,237 @@
...
+```
+
+Parameters:
+- `minimum-gas-prices`: The minimum gas price (in this example we use `0.005ubbn`)
+   that your node will accept for transactions. Transactions with lower gas 
+   prices will be rejected.
+- `iavl-cache-size`: Set to 0 to disable the IAVL tree caching. This reduces 
+   memory usage.
+- `iavl-disable-fastnode`: Set to true to disable the fast node feature. This
+   reduces memory usage.
```
Comment (Lazar955) [inline bbn-test-5/babylon-node/README.md:102]: tho, RPC performance is heavily impacted with no cache and no fastnode cache

File: bbn-test-5/babylon-node/README.md (hunk 16 above)
Comment (Lazar955) [reply bbn-test-5/babylon-node/README.md:86]: Otherwise RPC performance is impacted

**Review by Lazar955 — APPROVED** [review]
lgtm! Left some comments

**Review by Lazar955 — DISMISSED** [review]

File: bbn-test-5/babylon-node/README.md [hunk 18]
```diff
@@ -0,0 +1,242 @@
...
+quickly catch up to the current state without downloading and verifying the 
+entire blockchain history. To find the state-sync server from our 
+[networks homepage](../README.md).
+
+To utilize state sync, you'll need to update a few flags in your `config.toml`:
+```shell
+[statesync]
+enable = true
+
+rpc_servers = "Y"
```
Comment (maiquanghiep) [inline bbn-test-5/babylon-node/README.md:187]: Should we include the details of acquiring `trust_height` and `trust_hash` here or in the `networks hompage`? 
While the `rpc_servers` can be found in the network page, commands to get the  `trust_height` and `trust_hash` can be put here

**Review by filippos47 — APPROVED** [review]

**Review by vitsalis — APPROVED** [review]

**Pull request merged**